### PR TITLE
Add CreateCampaign Component

### PR DESCRIPTION
**Pull request by Retttro**
# Reviewers
- [ ] @okohuch
- [ ] @Antoniozinchenko 
- [x] @anastasiabalaban

# Jira task

# Description

Add CreateCampaign Component

# Checklist:

- [x] PR is approved by 1 team mate
- [ ] PR is approved by 2 experts

### CODE REVIEW

File: src/components/campaigns/CreateCampaign.js [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+  render() {
+    return (
+      <Modal trigger={<Button>Create Campaign</Button>} closeIcon>
+        <Modal.Header>
+          <h1>Create Campaign</h1>
+        </Modal.Header>
+        <Segment>
+          <Form onSubmit={this.handleSubmit}>
+            <Form.Group widths='equal'>
+              <Form.Input label="Title" type="text" value={this.state.value} onChange={this.handleTitleChange} />
```
Comment (anastasiabalaban) [inline src/components/campaigns/CreateCampaign.js]: You can write:
onChange={event => this.setState({title: event.target.value})}

Then you do not need a function handleTitleChange(event) and this.handleTitleChange = this.handleTitleChange.bind(this);

File: src/components/campaigns/CreateCampaign.js [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+              <Form.Input label="Needed Amount" type="text" value={this.state.value} onChange={this.handleNeededAmountChange} />
```
Comment (anastasiabalaban) [inline src/components/campaigns/CreateCampaign.js]: Same here

File: src/components/campaigns/CreateCampaign.js [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+            </Form.Group>
+            <Form.TextArea label="Description" value={this.state.value} onChange={this.handleDescriptionChange} />
```
Comment (anastasiabalaban) [inline src/components/campaigns/CreateCampaign.js]: Same here

**Review by anastasiabalaban — APPROVED** [review]

File: src/components/campaigns/CreateCampaign.js [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+    event.preventDefault();
+    const campaign = {
+      id: this.props.items.length,
+      title: this.state.title,
+      description: this.state.description,
+      currentAmount: 10,
+      neededAmount: this.state.neededAmount,
+      image: 'http://via.placeholder.com/350x150'
+    }
+    dbCampaigns.doCreateCampaigns(campaign.id, campaign.title,campaign.description,campaign.currentAmount,campaign.neededAmount, campaign.image)
```
Comment (Antoniozinchenko) [inline src/components/campaigns/CreateCampaign.js:29]: Close modal on .then()

File: src/components/campaigns/CreateCampaign.js [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+          <h1>Create Campaign</h1>
+        </Modal.Header>
+        <Segment>
+          <Form onSubmit={this.handleSubmit}>
+            <Form.Group widths='equal'>
+              <Form.Input label="Title" type="text" value={this.state.value} onChange={event => this.setState({title: event.target.value})} />
+              <Form.Input label="Needed Amount" type="text" value={this.state.value} onChange={event => this.setState({neededAmount: event.target.value})} />
+            </Form.Group>
+            <Form.TextArea label="Description" value={this.state.value} onChange={event => this.setState({description: event.target.value})} />
+            <Button type="submit">Submit</Button>
```
Comment (Antoniozinchenko) [inline src/components/campaigns/CreateCampaign.js]: add disabled prop if required inputs are empty

File: src/components/campaigns/CreateCampaign.js [hunk 6]
```diff
@@ -0,0 +1,49 @@
+import React from 'react';
```
Comment (Antoniozinchenko) [inline src/components/campaigns/CreateCampaign.js]: import React, { Component } from 'react';
use Component instead of React.Component

**Review by Antoniozinchenko — CHANGES_REQUESTED** [review]

File: src/components/campaigns/CreateCampaign.js [hunk 7]
```diff
@@ -23,11 +27,22 @@ class CreateCampaign extends React.Component {
...
+  close(){
+    this.setState({ showModal: false });
   }
 
   render() {
+    const {title, description, neededAmount} = this.state;
+    const isEnabled = 
+            title.length > 0 &&
+            description.length > 0 &&
+            neededAmount.length > 0;
```
Comment (Antoniozinchenko) [inline src/components/campaigns/CreateCampaign.js]: const isEnabled = !(title && description && neededAmount)
or
const isEnabled = title. length && description length && neededAmount length

**Review by Antoniozinchenko — APPROVED** [review]

**Pull request merged**